lib/bracket-matcher.coffee

```diff
@@ -218,3 +224,6 @@ class BracketMatcher
 
   unsubscribe: ->
     @subscriptions.dispose()
+
+  getScopedSetting: (key) ->
+    atom.config.get(key, scope: @editor.getLastCursor().getScopeDescriptor())
```

Comment:
Using a file with 1000 lines reading `Hello World`. Wrap every `Hello World` in brackets.
__before__: 4397.7 ms for `wrapSelectionInBrackets` (6 ms total for `atom.config.get`).
__after__: 6509.5 ms for `wrapSelectionInBrackets` (27.6 ms total for `atom.config.get`).

About my fix: I moved the config lookup of `wrapSelectionsInBrackets` into `@editor.mutateSelectedText` with `selection.cursor.getScopeDescriptor()`.
Right now, I just `return unless` (similar to how you drop [empty selections](https://github.com/atom/bracket-matcher/blob/master/lib/bracket-matcher.coffee#L160)). Is that what we want ?

If performance is a problem, I would change it to `getRootDescriptor()`.